packages/docusaurus-theme-classic/src/theme/NavbarItem/index.tsx

```diff
@@ -43,18 +43,23 @@ const getNavbarItemComponent = (type: NavbarItemComponentType) => {
   return navbarItemComponentFn();
 };
 
-function getComponentType({type, ...props}: Props): NavbarItemComponentType {
+function getComponentType(
+  type: string | undefined,
+  isDropdown: boolean,
+): NavbarItemComponentType {
   // Backward compatibility: navbar item with no type set
   // but containing dropdown items should use the type "dropdown"
   if (!type || type === 'default') {
-    const isDropdown = (props as DropdownNavbarItemProps).items !== undefined;
     return isDropdown ? 'dropdown' : 'default';
   }
   return type as NavbarItemComponentType;
 }
 
-export default function NavbarItem(props: Props): JSX.Element {
-  const componentType = getComponentType(props);
+export default function NavbarItem({type, ...props}: Props): JSX.Element {
+  const componentType = getComponentType(
+    type,
+    (props as DropdownNavbarItemProps).items !== undefined,
+  );
```

Comment:
I don't understand why you changed this, this makes the code more complex and does not seem to change the behavior?

Reply:
Since I've destructed `type`, I thought I could make this function bit simpler by making `isDropdown` as boolean param, isn't it?

Reply:
Looks less readable to me but definitively works.

I'd found this more readable if you didn't destructure type and used something like `<NavbarItemComponent {...props} type={undefined} />`